### PR TITLE
Update IDL tests for WebNN API.

### PR DESCRIPTION
**Pull request by BruceDai**
According to [latest WebNN API Spec](https://webmachinelearning.github.io/webnn/), this PR is to

1. Support deviceType for `navigator.ml.createContext()` interface
2. Make `buildAsync()` asynchronous execution run both in Window and in Worker, make `build()` synchronous execution only run in Worker

@dontcallmedom @Honry PTAL, thanks.

### CODE REVIEW

File: webnn/resources/utils.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+
+const ExecutionArray = ['async', 'sync'];
```
Comment (dontcallmedom) [inline webnn/resources/utils.js:3]: I think I would simply inline the two arrays in webnn/idlharness

File: webnn/idlharness.https.any.js [hunk 2]
```diff
@@ -9,8 +10,8 @@
 
 idl_test(
   ['webnn'],
-  ['html', 'WebIDL', 'webgl1', 'webgpu'],
-  idl_array => {
+  ['html', 'webidl', 'webgl1', 'webgpu'],
```
Comment (dontcallmedom) [inline webnn/idlharness.https.any.js:13]:
```suggestion
  ['html', 'webidl', 'webgpu'],
```
(webnn no longer depends on WebGL)

File: webnn/resources/utils.js (hunk 1 above)
Comment (BruceDai) [reply webnn/resources/utils.js:3]: Hi @dontcallmedom This `ExecutionArray` is also for operations tests, so I define it in this `utils.js` file.

File: webnn/idlharness.https.any.js (hunk 2 above)
Comment (BruceDai) [reply webnn/idlharness.https.any.js]: I've updated this PR, please take another look, thanks.

**Review by dontcallmedom — APPROVED** [review]

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Pull request merged**